examples/asyncio/asyncio-server.py

```diff
@@ -40,18 +43,26 @@ def connection_made(self, transport: asyncio.Transport):
         self.transport.write(self.conn.data_to_send())
 
     def data_received(self, data: bytes):
-        events = self.conn.receive_data(data)
-        self.transport.write(self.conn.data_to_send())
-
-        for event in events:
-            if isinstance(event, RequestReceived):
-                self.request_received(event.headers, event.stream_id)
-            elif isinstance(event, DataReceived):
-                self.receive_data(event.data, event.stream_id)
-            elif isinstance(event, StreamEnded):
-                self.stream_complete(event.stream_id)
-
+        try:
+            events = self.conn.receive_data(data)
+        except ProtocolError as e:
+            self.conn.close_connection(error_code=e.error_code)
```

Comment:
This should not be necessary. A `ProtocolError` will already have written the GOAWAY frame, so you can just do the write/close below.